[EDU-4528] feat: add live streaming use case - architecture (#934)
aziontech/docs: src/content/docs/en/pages/architectures/live-streaming-delivery.mdx

---
title: Optimize the distribution of video content with live streaming delivery
description: This architecture leverages advanced features like real-time transcoding, adaptive bitrate streaming, and intelligent caching, enabling seamless experiences for viewers.
meta_tags: architecture, content delivery, hls, edge computing
namespace: docs_arch_live_streaming_delivery
permalink: /documentation/architecture/live-streaming-delivery/live-streaming-delivery-with-hls/
---

> Edge computing transforms [live streaming delivery](ADD LINK***) by optimizing the distribution of high-quality video content. This architecture leverages advanced features like real-time transcoding and adaptive bitrate streaming, which, along with intelligent caching, enable seamless experiences for viewers. The distributed nature of edge computing allows for content to be processed and stored closer to the user, reducing latency and bandwidth constraints, and ensuring a high-quality stream even during periods of high demand.

Author: ```suggestion
Edge computing transforms [live streaming delivery](https://www.azion.com/en/solutions/live-streaming/) by optimizing the distribution of high-quality video content. This architecture leverages advanced features like real-time transcoding and adaptive bitrate streaming, which, along with intelligent caching, enable seamless experiences for viewers. The distributed nature of edge computing allows for content to be processed and stored closer to the user, reducing latency and bandwidth constraints, and ensuring a high-quality stream even during periods of high demand.
```

This solution is ideal for content providers who require high fidelity and low latency for live events, e-sports, and educational content. Utilizing a network of edge locations, the platform ensures efficient delivery and scalability, while providing enhanced security features for stream protection.

---

## Live Streaming Delivery Architecture Diagram

![Architecture diagram](assets/docs/images/uploads/diagram-live-streaming-delivery.png)

### Live Streaming Delivery Dataflow

1. The client (player) initiates a request to access a live stream.
2. The request reaches the healthiest edge node, which is part of Azion's globally distributed Edge Network.
- The edge application processes the request through the Rules Engine, applying security measures, origin configurations, and caching policies.
3. If the requested stream content is cached in the Edge Cache (recently received live content), it's immediately delivered to the client.
4. If the content isn't cached, the request is forwarded to the ingest component within the Azion Edge Platform.

> 5.  The source generates the content (video/audio).

Author: ```suggestion
5. The source generates the content (video/audio).
```

6. The Ingest (Live Ingest for RTMP or Tiered Cache for HLS) receives the encoded transmission, processes it, and answers to the edge node in HLS format (the same package is reutilized for all the edge nodes).
7. The transcoded stream is cached in the edge cache for subsequent high-demand simultaneous client requests.
8. The Rules Engine may be programmed to intercept errors, apply additional rules, process HTTP response headers, etc.
9. The processed transmission is delivered to the client, completing the live-streaming process. The user plays the content.

### Components

- [Edge Application](/en/documentation/products/build/edge-application/) allows you to build applications on top of the Azion Edge Platform and deliver your content.
  - [Edge Cache](/en/documentation/products/build/edge-application/edge-cache/) enables you to configure how your application caches content to reduce latency and increase transfer rates.
  - The **Ingest** component receives signal from encoders and distributes content in HLS format. It works as the origin for the edge application.

---

>   ## Implementation

Author: ```suggestion
## Implementation
```

    - Then, create a cache policy for the playlist.
        - Repeat the previous process, but this time set a **Maximum TTL** of `0` for **Browser Cache Settings** and `5` for **Edge Cache Settings**. Define **Advanced Cache Key** settings too.
    - Still on the application page, open the **Rules Engine** tab:
        - Create a rule for the chunks: select `if ${uri}` *matches* `.\*.ts` as criteria and **Set Cache Policy** as behavior, adding the policy for the chunks you created in the previous step.
        - Create a rule for the playlist: define `if ${uri}` *matches* `\*.m3u8` as criteria and **Set Cache Policy** as behavior, adding the policy for the playlist you created in the previous step.
4.  Configure your source and encoder pointing to Azion Ingest DNS entries.
5.  Stream your content via Azion Edge Platform.

---

## Related docs

- [Azion Edge Network](https://www.azion.com/en/products/edge-network/)
- [Cache Settings](/en/documentation/products/build/edge-application/cache-settings/)

> - [Content delivery - Architecture](ADD LINK***)

Author: ```suggestion
- [Content delivery architecture](/en/documentation/architectures/edge-application/content-delivery/)
```
